src/platform/plugins/shared/discover/public/context_awareness/profile_providers/observability/observability_document_profile/document_profile/profile.ts

```diff
@@ -17,17 +17,16 @@ import { createGetDocViewer } from './accessors';
 const OBSERVABILITY_GENERIC_DOCUMENT_PROFILE_ID = 'observability-generic-document-profile';
 
 export const createObservabilityGenericDocumentProfileProvider = ({
-  tracesContextService,
-  apmErrorsContextService,
+  apmContextService,
   logsContextService,
 }: ProfileProviderServices): DocumentProfileProvider => ({
   profileId: OBSERVABILITY_GENERIC_DOCUMENT_PROFILE_ID,
   restrictedToProductFeature: TRACES_PRODUCT_FEATURE_ID,
   profile: {
     getDocViewer: createGetDocViewer({
       apm: {
-        errors: apmErrorsContextService.getErrorsIndexPattern(),
-        traces: tracesContextService.getAllTracesIndexPattern(),
+        errors: apmContextService.errorsService.getErrorsIndexPattern(),
```

Comment:
note: Can we avoid fetching the indices on every document change? They aren’t expected to change that frequently!

Reply:
If not expected to change, then maybe we can cache the response once it has been fetched. But that should be managed by the context service

Reply:
As an example: the indices constant could be elevated to the module scope, so then when you call `apmSourcesAccess.getApmIndices()` you do the following:

```ts
indices ??= await apmSourcesAccess.getApmIndices();
```

If uninitialised, it calls and awaits the response, then putting it into `indices` and further calls don't trigger it. There *might* be a race condition here though with this approach, so probably needs testing/tweaking.